pkg/repositories/gormimpl/task_repo.go

```diff
@@ -21,14 +22,32 @@ type TaskRepo struct {
 	metrics          gormMetrics
 }
 
-func (r *TaskRepo) Create(ctx context.Context, input models.Task) error {
+func (r *TaskRepo) Create(_ context.Context, input models.Task, descriptionEntity *models.DescriptionEntity) error {
 	timer := r.metrics.CreateDuration.Start()
-	tx := r.db.Omit("id").Create(&input)
+	err := r.db.Transaction(func(_ *gorm.DB) error {
+		if descriptionEntity == nil {
+			tx := r.db.Omit("id").Create(&input)
+			if tx.Error != nil {
+				return r.errorTransformer.ToFlyteAdminError(tx.Error)
+			}
+			return nil
+		}
+		tx := r.db.Omit("id").Create(descriptionEntity)
+		if tx.Error != nil {
+			return r.errorTransformer.ToFlyteAdminError(tx.Error)
+		}
+		r.db.Last(descriptionEntity)
```

Comment:
you can use [FirstOrCreate](https://gorm.io/docs/advanced_query.html#FirstOrCreate) when you create the description entity to get is db id and avoid this lookup